### ISSUE TITLE
Build hierarchy of landuse polygons.

### ISSUE THREAD

**Issue by zerebubuth** [mapzen/vector-datasource#446]
A lot of the time, it's not possible to have a "static" hierarchy (e.g: all parks draw on top of all forests) between landuse polygons. Instead, we can use their geometry to determine their relationship; a featyre that is fully within another should probably draw on top of it.

We can build a "dynamic" hierarchy per-tile to produce a per-tile `order` value for each polygon which respects these relationships.

Note that there might be drawbacks to this approach; for example it would be possible for a polygon which crosses tile boundaries to get a different `order` in each tile, making it more likely that seaming artefacts would be visible. Many polygons won't have a simple within/contains relationship, and will have something more complex (crossing / mostly-overlapping). For those, there won't be a unique `order`, and we'll still have to tie-break to some arbitrary values.

**Comment by nvkelso** [comment]
Water that needs to displayed on top of landuse also needs to be in the mix (either with a data value set on export, or a conventional value communicated in a backchannel to the stylesheet).

**Comment by nvkelso** [comment]
We should also consider landuse versus nature (landcover) here.

**Comment by zerebubuth** [comment]
Yup, and I'm sure there's lots of cases where `nature` and landcover are very different. We'll have to take a look over what data is available and sort out a sensible set of defaults with various overrides if we want to start a landcover layer (or, rather, separate out landcover from the existing landuse layer).

**Comment by nvkelso** [comment]
This is also affecting the `landuse_kind` road intersection – discovered when setting up Tailgate Mania targeted editing post.

**Comment by nvkelso** [comment]
**Related:** The grass areas in the SFO Airport south of San Francisco show up green "park". It'd be nice to know this "landcover" natural feature is "in" an airport and downplay it:

- https://whosonfirst.mapzen.com/lampzen/maps/359

![359-vupraeulblpy](https://cloud.githubusercontent.com/assets/853051/16629274/0ee99816-436a-11e6-8366-e4ed0787e037.png)
